lib/client/api.go

```diff
@@ -2867,13 +2860,7 @@ func (tc *TeleportClient) ConnectToCluster(ctx context.Context) (_ *ClusterClien
 		),
 	)
 	defer func() {
-		if err != nil {
-			// The error is unwrapped here so that the error reported is any underlying
-			// error and not a trace.TraceErr.
-			span.RecordError(trace.Unwrap(err))
```

Comment:
Have we just been lazy, or is this a distinction that we have to make in the places where we currently do `trace.Unwrap` versus the places where we don't?

Reply:
Based on the conversation I had with Tim, it sounds like the places where we didn't currently unwrap are doing it wrong. If it's wrapped, we just always see the error type in the trace view as "*TraceErr" rather than something more useful. Whilst we might lose some wrapping information, the actual context from the traces themselves is equivalent or better.